libgalois/include/katana/PropertyGraph.h

```
@@ -144,6 +147,28 @@ class KATANA_EXPORT PropertyGraph {
     return rdg_.edge_properties();
   }
 
+  // recreate indexes from json
+  void recreate_node_property_indexes() {
+    node_property_indexes_column_name =
+        rdg_.get_node_property_indexes_column_name();
+    for (const std::string& column_name : node_property_indexes_column_name) {
+      auto result = MakeNodeIndex(column_name);
+      if (!result) {
+        return (void)result.error();
+      }
+    }
+  }
+  void recreate_edge_property_indexes() {
+    edge_property_indexes_column_name =
+        rdg_.get_edge_property_indexes_column_name();
+    for (const std::string& column_name : edge_property_indexes_column_name) {
+      auto result = MakeEdgeIndex(column_name);
+      if (!result) {
+        return (void)result.error();
```

Better to return this error than swallowing it.